pkg/controller/nodelifecycle/scheduler/taint_manager_test.go

```diff
@@ -109,60 +111,62 @@ func createNoExecuteTaint(index int) v1.Taint {
 }
 
 // Waits for delete actions to appear and then returns them
-func getDeleteActions(fakeClientset *fake.Clientset) []clienttesting.Action {
-	var actions []clienttesting.Action
-main:
-	for ; ; actions = fakeClientset.Actions() {
-		for _, action := range actions {
-			if action.GetVerb() == "delete" {
-				break main
+func waitForDeleteActionsForPods(fakeClientset *fake.Clientset) bool {
+	err := wait.PollImmediate(10*time.Millisecond, 5*time.Second, func() (bool, error) {
+		for actions := fakeClientset.Actions(); ; actions = fakeClientset.Actions() {
+			for _, action := range actions {
+				if action.GetVerb() == "delete" && action.GetResource().Resource == "pods" {
+					return true, nil
+				}
 			}
 		}
+	})
+
+	if err != nil {
+		return false
 	}
-	return actions
+	return true
 }
 
 // Waits for delete actions to appear and checks if pod names
 // are equal to those stated in actions. When there appeared
 // all the actions with stated pod names then those actions returned
-func getDeleteActionsWithCertainPods(fakeClientset *fake.Clientset, podNames []string) []clienttesting.Action {
-	var actions []clienttesting.Action
-
-	var podActionsReceived []string
-	isEqual := func(a []string, b []string) bool {
-	main:
-		for _, valueA := range a {
-			for _, valueB := range b {
-				if valueA == valueB {
-					continue main
+func waitForDeleteActionsForCertainPods(fakeClientset *fake.Clientset, podNames sets.String) bool {
+	err := wait.PollImmediate(10*time.Millisecond, 5*time.Second, func() (bool, error) {
+		for actions := fakeClientset.Actions(); ; actions = fakeClientset.Actions() {
+			for _, action := range actions {
+				deleteAction, ok := action.(clienttesting.DeleteActionImpl)
+				if !ok {
+					continue
 				}
-			}
-			return false
-		}
-		return true
-	}
-main:
-	for ; ; actions = fakeClientset.Actions() {
-	secondary:
-		for _, action := range actions {
-			deleteAction, ok := action.(clienttesting.DeleteActionImpl)
-			if !ok {
-				continue secondary
-			}
-			podActionsReceived = append(podActionsReceived, deleteAction.GetName())
 
-			if isEqual(podActionsReceived, podNames) {
-				break main
+				if podNames.Has(deleteAction.GetName()) {
+					return true, nil
```

Comment:
Here you return from the function as soon as any of the pod names are encountered. Instead, compute the set of encountered pod names, and compare the entire sets.